CardPayments/src/main/java/com/paypal/android/cardpayments/CardVaultResult.kt

```diff
@@ -1,18 +1,19 @@
 package com.paypal.android.cardpayments
 
-import android.os.Parcelable
-import kotlinx.parcelize.Parcelize
-
 /**
  * @suppress
  *
  * A result returned by [CardClient] when an a successful vault occurs.
  *
  * @param setupTokenId the id for the setup token that was recently updated
  * @param status the status of the updated setup token
+ * @param authChallenge an authentication challenge; will be non-null when 3DS authentication
+ * is required to complete a vault
  */
-@Parcelize
+// NEXT MAJOR VERSION: make `CardVaultResult` constructor private
 data class CardVaultResult(
     val setupTokenId: String,
-    val status: String
-) : Parcelable
+    val status: String,
+    // NOTE: This technically needs to null by default to prevent a breaking change
+    val authChallenge: CardAuthChallenge? = null
```

Comment:
If the SDK is creating the instances of `CardVaultResult`, would having `authChallenge` as non null be a breaking change. It might just add a new property to the result object that merchants could reference in the new SDK version.

Reply:
Yeah it's only breaking since the constructor is public by default, though technically we could treat it as non-breaking since we don't expect merchants to ever instantiate these internal types.

Next major version we should mark constructors as internal where we can, or at least restrict them with `LIBRARY_GROUP` annotations to keep them available for merchant unit tests.

Reply:
Gotcha, if we treat it as a non-breaking change, could we make `CardAuthChallenge` non-null? This would prevent the merchant from having to check nullability.

This is maybe a larger question, if we make breaking changes on classes that should be treated as internal, do we need treat them as a major version breaking change?

Reply:
Oh actually looking back at it my comment is incorrect, my bad. Semantically this value should be `null` when 3DS is not required. It's the default argument that's required to prevent a strict breaking change.

Reply:
Ah gotcha, all good then!